localstack/aws/client.py

```diff
@@ -92,4 +118,51 @@ def parse_response(operation: OperationModel, response: Response) -> ServiceResp
         response_dict["body"] = response.data
 
     parser = create_response_parser(operation.service_model.protocol)
-    return parser.parse(response_dict, operation.output_shape)
+    parsed_response = parser.parse(response_dict, operation.output_shape)
+
+    if response.status_code >= 301:
+        # Add possible additional error shape members
+        _add_modeled_error_fields(response_dict, parsed_response, operation, parser)
+
+    if not include_response_metadata:
+        parsed_response.pop("ResponseMetadata", None)
+
+    return parsed_response
+
+
+def parse_service_exception(
+    response: Response, parsed_response: Dict
+) -> Optional[ServiceException]:
+    """
+    Creates a ServiceException from a parsed response (one that botocore would return).
+    It does not automatically raise the exception (see #raise_service_exception).
+    :param response: Un-parsed response
+    :param parsed_response: Parsed response
+    :return: ServiceException or None (if it's not an error response)
+    """
+    if response.status_code < 301 or "Error" not in parsed_response:
```

Comment:
actually did a quick check on this, and turns out these operations are success responses but would from my understanding pass this if statement.

```
amplifybackend CloneBackend CloneBackendResponse Error
amplifybackend CreateBackend CreateBackendResponse Error
amplifybackend CreateBackendAPI CreateBackendAPIResponse Error
amplifybackend CreateBackendAuth CreateBackendAuthResponse Error
amplifybackend DeleteBackend DeleteBackendResponse Error
amplifybackend DeleteBackendAPI DeleteBackendAPIResponse Error
amplifybackend DeleteBackendAuth DeleteBackendAuthResponse Error
amplifybackend GenerateBackendAPIModels GenerateBackendAPIModelsResponse Error
amplifybackend GetBackend GetBackendResponse Error
amplifybackend GetBackendAPI GetBackendAPIResponse Error
amplifybackend GetBackendAuth GetBackendAuthResponse Error
amplifybackend GetBackendJob GetBackendJobResponse Error
amplifybackend ImportBackendAuth ImportBackendAuthResponse Error
amplifybackend RemoveAllBackends RemoveAllBackendsResponse Error
amplifybackend RemoveBackendConfig RemoveBackendConfigResponse Error
amplifybackend UpdateBackendAPI UpdateBackendAPIResponse Error
amplifybackend UpdateBackendAuth UpdateBackendAuthResponse Error
amplifybackend UpdateBackendConfig UpdateBackendConfigResponse Error
amplifybackend UpdateBackendJob UpdateBackendJobResponse Error
glue CheckSchemaVersionValidity CheckSchemaVersionValidityResponse Error
iam GetServiceLastAccessedDetails GetServiceLastAccessedDetailsResponse Error
iam GetServiceLastAccessedDetailsWithEntities GetServiceLastAccessedDetailsWithEntitiesResponse Error
lakeformation GetQueryState GetQueryStateResponse Error
migration-hub-refactor-spaces GetApplication GetApplicationResponse Error
migration-hub-refactor-spaces GetEnvironment GetEnvironmentResponse Error
migration-hub-refactor-spaces GetRoute GetRouteResponse Error
migration-hub-refactor-spaces GetService GetServiceResponse Error
redshift-data DescribeStatement DescribeStatementResponse Error
stepfunctions StartSyncExecution StartSyncExecutionOutput error
```

here's the script i used:
```python
from botocore.model import StructureShape

from localstack.aws.spec import iterate_service_operations


def main():
    for service, operation in iterate_service_operations():
        if not operation.output_shape:
            continue
        shape: StructureShape = operation.output_shape
        for member_name, member_shape in shape.members.items():
            if member_name.lower() == "error":
                print(service.service_name, operation.name, shape.name, member_name)


if __name__ == '__main__':
    main()
```

btw it would be great if we could add this method to `localstack.aws.spec` :grin: 
```python

def iterate_service_operations() -> Generator[Tuple[ServiceModel, OperationModel], None, None]:
    """
    Returns one record per operation in the AWS service spec, where the first item is the service model the operation
    belongs to, and the second is the operation model.

    :return: an iterable
    """
    for service in list_services():
        for op_name in service.operation_names:
            yield service, service.operation_model(op_name)
```